models/app.models.js

```diff
@@ -53,27 +53,43 @@ exports.selectCommentsByArticleId = (article_id) => {
 
 exports.insertCommentByArticleId = (article_id, commentToPost) => {
 
+    // Check for any bad requests and reject the promise
+    if (!commentToPost.username || !commentToPost.body) return Promise.reject({status: 400, msg: "Bad request"})
+    
+    // Begin my checking for whether the username exists
     return db.query(`
-    SELECT DISTINCT author FROM articles WHERE article_id = $1`, [article_id])
-    .then(result => {
-        if (result.rows.length === 0) return Promise.reject({status: 404, msg: "Not found"})
-        return author = result.rows[0].author
+    SELECT * FROM users
+    WHERE username = $1
+    `, [commentToPost.username])
+    .then(username => {
+        // If username does not exist, return a rejected promise of 'not found'
+        if (!username.rows[0]) return Promise.reject({status: 404, msg: "Not found"})
```

Comment:
Your seed function will not allow a post from a user that doesn't exist, you should get an error code that you can grab in your error handlers to avoid two calls to the database. Alternatively you could implement a check exists function for this.